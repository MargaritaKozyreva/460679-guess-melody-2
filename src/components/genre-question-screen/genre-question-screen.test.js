import React from "React";
import GenreQuestionScreen from "./genre-question-screen.jsx";
import renderer from "react-test-renderer";

describe(`GenreQuestionScreen initiall`, () => {

  const formSubmitHandler = jest.fn();
  const checkboxCheckedHandler = jest.fn();

  let isLoading = true;

  const mockProps = {
    question: {
      answers: [
        {
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
          genre: `rock`
        },
        {
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
          genre: `pop`
        },
        {
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
          genre: `jazz`
        },
        {
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
          genre: `rock`
        }
      ],
      genre: `test`
    },
    screenIndex: 0,
    formSubmitHandler,
    checkboxCheckedHandler
  };

  it(`GenreQuestionScreen renders correctly`, () => {
    const tree = renderer.create(<GenreQuestionScreen {...mockProps}/>, {
      createNodeMock: (element) => {
        if (element.type === `audio`) {
          return {
            oncanplaythrough: () => {
              isLoading = false;
            }
          };
        }
        return null;
      }
    }).toJSON();
    expect(isLoading).toBe(true);
    expect(tree).toMatchSnapshot();
  });
});
