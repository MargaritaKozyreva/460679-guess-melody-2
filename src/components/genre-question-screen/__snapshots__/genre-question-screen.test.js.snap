// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreQuestionScreen initiall GenreQuestionScreen renders correctly 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="timer__value"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите 
      test
       треки
    </h2>
    <form
      className="game__tracks"
      onSubmit={[MockFunction]}
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={false}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="answer-0"
            name="undefined-0"
            onChange={[MockFunction]}
            type="checkbox"
            value="answer-0"
          />
          <label
            className="game__check"
            htmlFor="answer-0"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={false}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="answer-1"
            name="undefined-0"
            onChange={[MockFunction]}
            type="checkbox"
            value="answer-1"
          />
          <label
            className="game__check"
            htmlFor="answer-1"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={false}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="answer-2"
            name="undefined-0"
            onChange={[MockFunction]}
            type="checkbox"
            value="answer-2"
          />
          <label
            className="game__check"
            htmlFor="answer-2"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={false}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg"
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="answer-3"
            name="undefined-0"
            onChange={[MockFunction]}
            type="checkbox"
            value="answer-3"
          />
          <label
            className="game__check"
            htmlFor="answer-3"
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
